charts/tekton-pipelines/values.yaml

```diff
@@ -65,22 +65,36 @@ buildpacks:
           resources: {}
           workingDir: $(resources.inputs.app.path)
           script: |
-            #!/bin/bash
+            #!/usr/bin/env bash
             set -Eeo pipefail
 
-            [ -f package.json ] && {
-              echo "install node.js dependencies"
-
+            if [[ ! -f package.json ]]; then
+              echo "No package.json found, stepping out"
+            else
+              echo "Installing node.js dependencies"
               # set cache
               npm config set cache $(workspaces.source.path)/.npm --global
-              npm ci
-
-              (npm run build:$(params.environment) && rm -rf node_modules && echo "remove node_modules dir") || (echo "unable to build $(params.environment) exit_code: $?" && exit 1)
+              if ! npm ci; then
+                EXIT_CODE=$?
+                echo "unable to install dependencies, "
+                if [[ $EXIT_CODE == 143 ]]; then
+                  echo "oom kill detected"
+                else
+                  echo "exit_code: $EXIT_CODE"
+                fi
+                exit 1
+              fi
+              echo "Building $(params.environment) target"
+              if ! npm run build:$(params.environment); then
+                echo "unable to build $(params.environment) exit_code: $?"
+                exit 1
+              fi
+              echo "Cleaning up"
+              rm -rf node_modules
+              echo "Preparing for buildpack build"
```

Comment:
Misha, 
1. let's restore original - as this block of code provides no additional value. If `npm ci` fails - entire pipelinerun will fail. So no need to detect etc.
2. Original code was short and concise - your new implementation is longer and does the same thing.

Reply:
>If npm ci fails - entire pipelinerun will fail. So no need to detect etc.

As practics shows it does not. https://tekton-staging.theproducebox.com/#/namespaces/ci/pipelineruns/producebox-frontend-staging-build-pipeline-run-djgfl?pipelineTask=buildpack-frontend&step=build-static

Reply:
@dmitry-mightydevops this is normal behavior
![image](https://user-images.githubusercontent.com/31823452/236132940-baac64be-9a8e-449f-a47e-84125696aeef.png)

Reply:
@MikhailKorol-saritasa pls give me instructions to replicate the failed build - so I can see how original implementation worked - pls create a sate in the code (separate git branch) in which the build will not work as expected - I will take a look closely.

Reply:
IMHO, new implementation is readable enough and adds value.